### PR TITLE
Update developer guide

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Addresses https://github.com/dotnet/tye/issues/540

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -13,4 +13,4 @@ while [[ -h $source ]]; do
 done
 
 scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
-"$scriptroot/eng/common/build.sh" --pack --build --restore $@
```
Comment (JunTaoLuo) [inline build.sh:16]: It's odd that --pack is specified in the default build.sh, we don't do this elsewhere. Is there a reason for this in this repo?

Comment (jkotalik) [reply build.sh:16]: not sure, but we'd probably need to update the azure-pipelines yaml to add -pack if we remove it.

Comment (JunTaoLuo) [reply build.sh:16]: azure-pipelines uses eng/common/cibuild.sh which passes --pack so it's not necessary to update anything. I took a look and it seems nothing else in the repo references or uses build.sh directly.

Comment (jkotalik) [reply build.sh:16]: Alright, update it then.

Comment (JunTaoLuo) [reply build.sh:16]: Update what?

Comment (jkotalik) [reply build.sh:16]: Oh, you already removed pack... 😄

**Pull request merged**